pkg/graveler/graveler_test.go

```diff
@@ -1737,12 +1748,15 @@ func TestGraveler_PreCommitHook(t *testing.T) {
 			if tt.hook {
 				g.SetHooksHandler(h)
 			}
+			repo := deepcopy.Copy(repository).(*graveler.RepositoryRecord)
```

Comment:
This is scary: AFAICU deepcopy.Copy can only copy exported (uppercased) fields.  It so happens that RepositoryRecord and Repository are both plain-data objects -- but there is no guarantee that this will be true.

I believe it would be safer and easier to understand if we just used a separate object repositoryRO or something.  I think it would also be safer.